onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2243,15 +2254,10 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
             return ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
                                    "TensorRT EP could not deserialize engine from cache: " + engine_cache_path);
           }
-        } else if (engine_decryption_enable_ && engine_cache_enable_ && !engine_file && !engine_update) {
+        } else if (engine_decryption_enable_ && engine_cache_enable_ && encrypted_engine_exists && !engine_update) {
           // Decrypt engine
-          size_t engine_size = 0;
```

Comment:
There is a compile error.
If the definition of engine_size is removed, than line2665 will fail.

see
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1102203&view=logs&j=7536d2cd-87d4-54fe-4891-bfbbf2741d83&t=66420422-c7d6-5f71-625c-4b7851c9b9ba

Reply:
Ah right, line 2265 should use encrypted_engine_size. I fixed it, it compiles now.